packages/logger/src/ConsolaStructuredLogger.ts

```diff
@@ -0,0 +1,136 @@
+import { ConsolaOptions, ConsolaReporter, createConsola } from "consola";
+
+import dotenv from "dotenv";
+import { LogLevel } from "./interfaces/LogLevel";
+import type {
+  LogData,
+  LoggerInterface,
+  Metadata,
+} from "./interfaces/LoggerInterface";
+import type { LoggerOptions } from "./interfaces/LoggerOptions";
+import type { StructuredLog } from "./interfaces/StructuredLog";
+
+// We do not want to load the .env in the browser and in the edge runtime
+if (typeof window === "undefined" && process.env.NEXT_RUNTIME !== "edge") {
+  dotenv.config();
+}
+
+interface ConsolaLoggerOptions
+  extends LoggerOptions,
+    Partial<Omit<ConsolaOptions, "level">> {}
+
+const createConsolaStructuredLogger = (
+  structuredLog: StructuredLog,
+  options: ConsolaLoggerOptions = {
+    level: "info",
+    includeStackTrace: true,
+  }
+): LoggerInterface => {
+  const levelMap: Record<LogLevel, number> = {
+    emergency: 0,
+    alert: 0,
+    critical: 0,
+    error: 0,
+    warning: 1,
+    notice: 2,
+    info: 3,
+    debug: 4,
+  };
+
+  const jsonReporter: ConsolaReporter = {
+    log: (logObject) => {
+      if (
+        process.env.NODE_ENV === "development" ||
+        typeof window !== "undefined"
+      ) {
+        console.log(logObject.args[0].structuredLog);
+      } else {
+        console.log(JSON.stringify(logObject.args[0].structuredLog));
+      }
+    },
+  };
+
+  const logger = createConsola({
+    level: levelMap?.[options.level] ?? levelMap.info,
+    reporters: [jsonReporter],
+  });
+
+  if (options.reporters) {
+    logger.setReporters(options.reporters);
+  }
+
+  const mapToConsolaLevel = (level: LogLevel): number => {
+    return levelMap?.[level] ?? levelMap.info; // Default to 'info' level
+  };
+
+  const logStructured = (
+    level: LogLevel,
+    logData: LogData,
+    metadata?: Metadata
+  ): void => {
+    const structuredLogEntry = structuredLog.createLog(
+      logData,
+      options,
+      level,
+      metadata
+    );
+
+    const consolaLevel = mapToConsolaLevel(level);
```

Comment:
```suggestion
    // A few LogLevel's map to the same ConsolaLevel, so we had to operate on ConsolaLevel's to DRY
    const consolaLevel = mapToConsolaLevel(level);
```

This line and switch made me think a minute about "why". Maybe we could add comment or some human-readable object like (I prefer latter, what about you?):
```ts
const CONSOLA_LEVELS = {
  ERROR: 0,
  WARN: 1,
};

// then
case ERROR:
logger.error({ structuredLog: structuredLogEntry });
        break;
```

Reply:
I have added a comment.